src/randoop/ExceptionCheck.java

```diff
@@ -0,0 +1,145 @@
+  }
+
+  /** Returns the class of the exception thrown**/
+  @Override
+  public String get_value() {
+    return exception.getClass().getName();
+  }
+
+  @Override
+  public String get_id() {
+    return "Throws exception @" + statementIndex;
+  }
+
+  /** return the offset of this statement in the sequence **/
+  @Override
+  public int get_stmt_no() {
+    return statementIndex;
+  }
+
+  /**
+   * The "try" half of the try-catch wrapper.
+   */
+  @Override
+  public String toCodeStringPreStatement() {
+    StringBuilder b = new StringBuilder();
+    b.append("// The following exception was thrown during execution." + Globals.lineSep);
+    b.append("// This behavior will recorded for regression testing." + Globals.lineSep);
+    b.append("try {" + Globals.lineSep + "  ");
+    return b.toString();
+  }
+  
+  /** The nearest visible superclass -- usually the argument itself. */
+  // TODO: handle general visibility based on Randoop command-line
+  // arguments, rather than hard-coding isPublic test.
+  public static Class<?> nearestVisibleSuperclass(Class<?> clazz) {
+    while (! Modifier.isPublic(clazz.getModifiers())) {
+      clazz = clazz.getSuperclass();
+    }
+    return clazz;
+  }
+
+  /**
+   * Returns the "catch" half of the try-catch wrapper.
+   * Catches the closest public superclass of the exception.
+   * Calls {@link this#appendCatchBehavior(StringBuilder, String)} to determine
+   * what code to include in the catch block.
+   */
+  @Override
+  public String toCodeStringPostStatement() {
+    StringBuilder b = new StringBuilder();
+    String exceptionClassName = getExceptionName();
+    if (exceptionClassName == null) {
+      exceptionClassName = "Exception";
+    }
+    appendTryBehavior(b, exceptionClassName);
+    if (Modifier.isPublic(exception.getClass().getModifiers())) {
+      b.append("} catch (" + exceptionClassName + " e) {" + Globals.lineSep);
+      b.append("  // Expected exception." + Globals.lineSep);
+      b.append("}" + Globals.lineSep);
+    } else {
+      // The exception type is private.  Catch the nearest public supertype.
+      Class<?> publicSuperClass = nearestVisibleSuperclass(exception.getClass());
+      String publicSuperClassName = publicSuperClass.getCanonicalName();
+      b.append("} catch (" + publicSuperClassName + " e) {" + Globals.lineSep);
+      b.append("  // Expected exception." + Globals.lineSep);
+      appendCatchBehavior(b, exceptionClassName);
+      b.append("}" + Globals.lineSep);
+    }      
+    return b.toString();
+  }
+  
+  /**
+   * Appends code for catch block behavior corresponding to expected exception.
+   * 
+   * @param b  the string builder to which code text is to be added
+   * @param exceptionClassName  the class name of the expected exception
+   */
+  protected abstract void appendCatchBehavior(StringBuilder b, String exceptionClassName);
+
+  /**
+   * Appends code to follow statement throwing expected exception in try block.
```

Comment:
It took me a little while to parse whether "follow statement" should be "following statement" or "follow the statement".  I think it's usually worth taking a few extra words to make the result unambigious.